rust/datafusion/src/logical_plan/mod.rs

```diff
@@ -21,2300 +21,21 @@
 //! Logical query plans can then be optimized and executed directly, or translated into
 //! physical query plans and executed.
 
-use std::fmt::{self, Debug, Display};
-use std::{any::Any, collections::HashMap, collections::HashSet, sync::Arc};
-
-use aggregates::{AccumulatorFunctionImplementation, StateTypeFunction};
-use arrow::{
-    compute::can_cast_types,
-    datatypes::{DataType, Field, Schema, SchemaRef},
-};
-
-use crate::datasource::parquet::ParquetTable;
-use crate::datasource::TableProvider;
-use crate::error::{DataFusionError, Result};
-use crate::{
-    datasource::csv::{CsvFile, CsvReadOptions},
-    physical_plan::udaf::AggregateUDF,
-    scalar::ScalarValue,
-};
-use crate::{
-    physical_plan::{
-        aggregates, expressions::binary_operator_data_type, functions, udf::ScalarUDF,
-    },
-    sql::parser::FileType,
-};
-use arrow::record_batch::RecordBatch;
-use functions::{ReturnTypeFunction, ScalarFunctionImplementation, Signature};
-
+mod builder;
```

Comment:
This module now  `pub use` s all `struct`, `enum` and `traits` that were public in this module so no client code should need to change.